src/data-model-providers/codegen/NodeConfigurationListener.cpp

```diff
@@ -0,0 +1,36 @@
+/*
+ *    Copyright (c) 2025 Project CHIP Authors
+ *    All rights reserved.
+ *
+ *    Licensed under the Apache License, Version 2.0 (the "License");
+ *    you may not use this file except in compliance with the License.
+ *    You may obtain a copy of the License at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *    Unless required by applicable law or agreed to in writing, software
+ *    distributed under the License is distributed on an "AS IS" BASIS,
+ *    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *    See the License for the specific language governing permissions and
+ *    limitations under the License.
+ */
+#include <data-model-providers/codegen/NodeConfigurationListener.h>
+
+namespace chip {
+namespace app {
+namespace {
+NodeConfigurationListener * gListener = nullptr;
+} // namespace
+
+NodeConfigurationListener * NodeConfigurationListener::GetNodeConfigurationListener()
```

Comment:
This state-keeping should not use globals, and should live in the DataModelProvider instead.

Reply:
Should be addressed in the latest commits